spec/openapi.yaml

```diff
@@ -53,29 +149,213 @@ paths:
+                  value: |
+                    {
+                      'apiVersion': '1.0',
+                      'data': {
+                          'category': 'Tutorial',
+                          'created_at': '2019-07-21 15:41:19',
+                          'downvotes': 0,
+                          'id': 2137,
+                          'languages': [],
+                          'last_updated': '2019-07-22 16:05:53',
+                          'name': 'Some, but not ALL of the knowledge',
+                          'notes': null,
+                          'paid': false,
+                          'times_clicked': 0,
+                          'upvotes': 1,
+                          'url': 'http://www.test.com'
+                      },
+                      'status': 'ok',
+                      'status_code': 200
+                    }
+      security:
+      - ApiKeyAuth: []
+  /resources/{id}/click:
+    put:
+      summary: UPDATE RESOURCE VISITS BY ID
+      description:
+        Increments the `times_clicked` property of a resource 
+      parameters:
+      - name: id
+        in: path
+        description: Resource ID
+        required: true
+        style: simple
+        explode: false
+        schema:
+          type: integer
+          format: int64
+      responses:
+        200:
+          description: Updated resource details
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/Resource'
               examples:
                 0:
                   value: |
-                    {"apiVersion":"1.0","data":{"category":"Books","created_at":"2019-06-15 17:55:32","downvotes":0,"id":3,"languages":[],"last_updated":"","name":"Free Tech Books","notes":"Focuses on general computer science concepts rather than a specific language","paid":false,"times_clicked":0,"upvotes":0,"url":"http://www.freetechbooks.com/"},"status":"ok","status_code":200}
+                    {
+                      'apiVersion': '1.0',
+                      'data': {
+                          'category': 'Tutorial',
+                          'created_at': '2019-07-21 15:41:19',
+                          'downvotes': 0,
+                          'id': 2137,
+                          'languages': [],
+                          'last_updated': '2019-07-22 16:05:53',
+                          'name': 'Some, but not ALL of the knowledge',
+                          'notes': null,
+                          'paid': false,
+                          'times_clicked': 3,
+                          'upvotes': 1,
+                          'url': 'http://www.test.com'
+                      },
+                      'status': 'ok',
+                      'status_code': 200
+                    }
+        404:
+          $ref: '#/components/responses/NotFound'
   /resources:
     get:
-      description: Gets list of all available resource entries
+      summary: RETRIEVE RESOURCES
```

Comment:
I would like to standardize how we name these endpoints. Paginated results such as this should be `List X`, in this case `List Resources`. The other endpoints should be `List Categories` and `List Languages`.

For endpoints that return a single resource, it should be `Get Resource`, `Get Language` and `Get Category`.

For endpoints that create, it should be `Create Resource`, etc.

For endpoints that update, it should be `Update Resource`, etc.